addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -810,7 +811,7 @@ var SnippetsMenu = Widget.extend({
                     }
                     $zone.before($drop);
                 }
-                if ($zone.next('.oe_drop_zone:visible').length === 0) {
+                if ((isDropClone && !$zone.prev('.oe_drop_zone:visible').length) || (!isDropClone && !$zone.next('.oe_drop_zone:visible, .oe_drop_clone').length)) {
```

Comment:
@qsm-odoo Actually, when it's a clone, we have have to check here if the previous is a drop zone or not.
Because if we have that in the dom (before the loop on selectorSiblings start)  :

<drop_zone>
<clone>
<snippet>
<snippet>
<snippet>
... etc.

a drop zone will be added after the clone and we dont want it
So I don't know if this code is more readable than my first code

Reply:
still working on it